src/QirTools/qirlib/src/emit/qir/instructions.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use super::{
+    array1d::{self, create_ctl_wrapper},
+    basic_values, calls,
+};
+use crate::{emit::Context, interop::Instruction};
+use inkwell::values::{BasicValueEnum, FunctionValue};
+use std::collections::HashMap;
+
+fn get_qubit<'ctx>(
+    name: &String,
+    qubits: &HashMap<String, BasicValueEnum<'ctx>>,
+) -> BasicValueEnum<'ctx> {
+    qubits.get(name).unwrap().to_owned()
```

Comment:
Does this `unwrap` fail when an invalid `name` is passed? If so, it may be worth using `?` instead to bubble that out into a user-actionable `Result` rather than `panic`ing when `name` is invalid.